Add caution message about reset container on each request (#16750)
symfony/symfony-docs: testing.rst

    :ref:`framework.test <reference-framework-test>` option is enabled).
    This means you can override the service entirely if you need to.

.. caution::

>     Before each request with the client, the client "resets" the container and recreates it from scratch.

Reviewer: We could be more specific here if it's worth it:
```suggestion
    Before each request, the client reboots the kernel, recreating the container from scratch.
```
?
Or maybe use `app` instead of `kernel`.

Author: `kernel` is a little bit more linked with `container` term, what do you think about it?

>     That gives each request an "isolated" environment because each request will create new service objects.

Reviewer: ```suggestion
    This ensures that every requests are "isolated" using "new" service objects.
```
?

>     But for example, when using the entity manager of Doctrine, all entities loaded with it are "lost"

Reviewer: ```suggestion
    Also, it means that entities loaded by Doctrine repositories will be "detached", so they will need to be
    refreshed by the manager or queried again from a repository.
```

    :ref:`framework.test <reference-framework-test>` option is enabled).
    This means you can override the service entirely if you need to.

.. caution::

>     Before each request, the client reboots the kernel, recreating the container from scratch.

Reviewer: Can you please be careful to break lines at 80 characters.

Author: Done

    This ensures that every requests are "isolated" using "new" service objects.
    Also, it means that entities loaded by Doctrine repositories will be "detached", so they will need to be
    refreshed by the manager or queried again from a repository.

>     when a new request is sent (because the entity manager was "reset"). You have to query again entities if you want

Reviewer: IMHO this line and the rest can be removed now.

Author: @HeahDude do you mean this one?

> when a new request is sent (because the entity manager was "reset"). You have to query again entities if you want

Reviewer: Yes!

Author: Done
